### PR TITLE
Site minor fixes

### LINKED ISSUE #1762: Site documentation needs some fixes
the site documentation needs some minor fixes and improvements, mainly:

- readability - separating long paragraphs, adding headlines
- fixing broken links
-  updating according to new fybrik versions (modules page has image that references mesh for data, vault plugins page use old blueprint fields)

### LINKED ISSUE #1765: The "IT Configuration policies"  web link from "Optimizer" web page returns 404
The "[IT Configuration policies](https://fybrik.io/v1.1/concepts/optimizer/config-policies)"  web link from "[Optimizer](https://fybrik.io/v1.1/concepts/optimizer/)" web page returns 404

### PR DESCRIPTION
**Pull request by aradhalevy**
site minor fixes, mainly to concepts section.

includes improving readability, fixing broken links and updating a few things who are not relevant to current fybrik version.  

note for reviewers: 95% of the changes are adding newline, re arranging paragraphs and fixing links and so on, and not changing contents.

/closes #1762  
/closes #1765

### CODE REVIEW

File: site/docs/concepts/architecture.md [hunk 1]
```diff
@@ -2,27 +2,53 @@
...
-current architecture of the Fybrik platform, running on top of Kubernetes. 
+current architecture of the Fybrik platform, running on top of Kubernetes.
 
 The core parts of Fybrik are based on Kubernetes controllers and Custom Resource Definitions (CRDs) in order to define its work items and reconcile the state of them.
 
 The primary interaction object for a data user is the `FybrikApplication` custom resource (1) where a user defines which data should be used for which purpose. The following chart and description describe the architecture and components of Fybrik relative to when they are used.
 
 ![Architecture](../static/architecture.png)
 
-Before the data user can perform any actions a data operator has to [install](../get-started/quickstart.md) Fybrik and modules.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:13]: I think we need this sentence here and not in the Modules section.

File: site/docs/concepts/architecture.md [hunk 2]
```diff
@@ -2,27 +2,53 @@
...
+## Modules
 
-Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application. The current abstraction supports 2 different [connectors](./connectors.md): one for data catalog (2) and one for policy manager (3). It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported. The data steward configures policies in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api) (6). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
+Before the data user can perform any actions a data operator has to [install](../get-started/quickstart.md) Fybrik and modules.  
+[Modules](./modules.md) (M) describe capabilities that can be included in a data plane.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:16]: Please add a link to the Modules section. https://fybrik.io/v1.1/concepts/modules/

File: site/docs/concepts/architecture.md [hunk 3]
```diff
@@ -2,27 +2,53 @@
...
+Modules may also perform actions to enforce data governance policy decisions, such as masking or redaction as examples.
 
-As data assets may reside in different clusters/clouds a `Blueprint` CRD is created for each cluster, containing the information regarding the services to be deployed or configured in the given cluster. Depending on the setup the `PlotterController` will use various methods to distribute the blueprints. In a multi cluster setup the default distribution implementation is using [Razee](http://razee.io) to control remote blueprints, but several multi-cloud tools
-could be used as a replacement. The `PlotterController` also collects statuses and distributes
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## Connectors
```
Comment (roytman) [inline site/docs/concepts/architecture.md:21]: please add a link to connectors

File: site/docs/concepts/architecture.md [hunk 4]
```diff
@@ -2,27 +2,53 @@
...
-could be used as a replacement. The `PlotterController` also collects statuses and distributes
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## Connectors 
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+The current abstraction supports 2 different [connectors](./connectors.md): one for data catalog (2) and one for policy manager (3). It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:24]: The current abstraction supports 4 different connectors. 

- data catalog
- data policy
- credentials
- CSP

P.S. There is inconsistency in the names.
here on the picture "Data Policy Connector" but at the Connectors page it is "Policy manager" connector
Credential connector -> Credential management connector
Need to add CSP to both pages.

File: site/docs/concepts/architecture.md [hunk 5]
```diff
@@ -2,27 +2,53 @@
...
+
+The data steward configures policies  in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api) (6). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
+
+## FybrikApplication
+Once the Modules and Connectors are installed, a developer can submit a `FybrikApplication` CRD to Kubernetes.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:29]: Once Fybrik is deployed and configured, include catalogs, modules and connectors, a data user can submit ....

File: site/docs/concepts/architecture.md [hunk 6]
```diff
@@ -2,27 +2,53 @@
...
+
+The controller uses the information provided in the `FybrikApplication`, to check with the data-governance policy manager if the data flow requested is allowed and whether restrictive actions such as masking or hashing have to be applied. Taking into account these governance actions, as well as application requirements, dataset specification, available infrastructure (5) and the [IT config policies](./config-policies.md) defined (4), the controller compiles a plotter (9).  
+
+## Plotter
+The plotter specifies a data plane connecting the application to the datasets it requires. More specifically, the plotter lists the modules to use, the capabilities required from these modules, the cluster on which each module should be deployed, as well as the flow of data between the asset and the workload through the chosen modules. 
+
+For the plotter to be optimal in terms of the defined optimization goals (a.k.a. IT config soft policies), the controller may use a [CSP-based optimizer](./optimizer.md) (7).
```
Comment (roytman) [inline site/docs/concepts/architecture.md:40]: you can add a link to IT policies

File: site/docs/concepts/architecture.md [hunk 7]
```diff
@@ -2,27 +2,53 @@
...
+
+## Blueprint
+As data assets may reside in different clusters/clouds a `Blueprint` CRD is created for each cluster, containing the information regarding the services to be deployed or configured in the given cluster. Depending on the setup the `PlotterController` will use various methods to distribute the blueprints. 
 
 A single [blueprint](../reference/crds.md#blueprint) contains the specification of all assets that shall be accessed in a single cluster by a single application.
 The `BlueprintController` makes sure that a blueprint can deploy all needed modules and tracks their status. Once e.g. an implicit-copy module finishes the copy the blueprint is also in a ready state.
-A read or write module is in ready state as soon as the proxy service such as the arrow-flight module is running. 
+A read or write module is in ready state as soon as the proxy service such as the arrow-flight module is running.
+
+In a multi cluster setup the default distribution implementation is using [Razee](http://razee.io) to control remote blueprints, but several multi-cloud tools could be used as a replacement.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:50]: but several multi-cluster tools could be used as a replacement.   -> but other multi-cluster tools could be used as a replacement.

File: site/overrides/home.html [hunk 8]
```diff
@@ -257,7 +257,7 @@ <h2>
...
             Control
         </h2>
-        <p>Control and optimize the workload behavior by orchestrating data functions and data mobility. Optimize performance using technologies such as caching, copies and more</p>
+        <p>Control and optimize the workload behavior by orchestrating data functions and data mobility. Optimize performance and cost using technologies such as caching, copies and more</p>
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/overrides/home.html:260]: The `cost` is added by me in this PR.
I think it shows that it wasn't in there before and I added it, if it is not the case let me know and I will try to solve it.

File: site/docs/concepts/taxonomy.md [hunk 9]
```diff
@@ -31,15 +36,13 @@ Different actors and components define the contents of different parts of the ta
...
-
-Default taxonomies are provided by fybrik, and are meant as a starting point on which to expand.
+As new capabilities, transforms, data types, and protocols are made available via FybrikModules, fybrik's module taxonomy must be updated.  Once updated these capabilities are available for use by other components, such as the Data Catalog and Data Governance Policy manager should they choose to leverage them.
 
 ## Validation Points
 
 Fybrik validates the structures and values it receives from all external components.  
 
-For interface components (FybrikApplication and FybrikModule), validation occurs when the resource is created, updated or deleted.  How validation errors are received depends on whether fybrik is deployed with webhooks or not.
+For interface components (`FybrikApplication` and `FybrikModule`), validation occurs when the resource is created, updated or deleted.  How validation errors are received depends on whether fybrik is deployed with [webhooks](https://en.wikipedia.org/wiki/Webhook) or not.
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/taxonomy.md]: Done

File: site/docs/concepts/taxonomy.md [hunk 10]
```diff
@@ -1,15 +1,20 @@
...
+some components that use taxonomy:
+
+* [FybrikApplication](../architecture/#fybrikapplication) yaml - information provided about the workload and the datasets.
+* Fybrik manager ([FybrikApplication controller](../architecture/#fybrikapplication)) - validates that the data is used in accord with the data governance policies and the IT config policies.
+* [Data catalog](../connectors/#data-catalog) - provides metadata about the asset.
+* [Data Governance Policy Manager](../connectors/#policy-manager) - defines the governance policies to follow.
+* [Config Policy Manager](./config-policies.md) - defines the IT policies to follow.
+* [FybrikModules](./modules.md) - describes capabilities that can be included in a data plane.
+
+Default taxonomies are provided by fybrik in a JSON file format, and are and are meant as a starting point on which to [expand](../../tasks/custom-taxonomy).
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/taxonomy.md]: changed to be 
and are meant as a starting point. Different actors can [expand] these.

File: site/docs/concepts/taxonomy.md [hunk 11]
```diff
@@ -1,15 +1,20 @@
 # Taxonomy
 
-Fybrik interacts with multiple external components, such as the data catalog, data governance policy manager, and modules.  In order for fybrik to orchestrate the data plane of a given workload it is essential that all the components involved use common terms.  For example, if a data governance policy refers to a particular transform it is crucial that the module implementing that transform refer to it in the same way, or for fybrik to be able to map between the disparate terms.
+A taxonomy defines the terms and related values that need to be commonly understood and supported across the components in the system.
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/taxonomy.md:3]: this change included only bringing this sentence before the rest: 

"A taxonomy defines the terms and related values that need to be commonly understood and supported across the components in the system." 

all the explanation is still there, Git wrongly marked it as deleted.

File: site/docs/concepts/modules.md [hunk 12]
```diff
@@ -38,8 +38,9 @@ kubectl apply -f https://raw.githubusercontent.com/fybrik/arrow-flight-module/ma
 ## When is a module used?
 
 There are three main data flows in which modules may be used:
-* Read - preparing data to be read and/or actually reading the data
-* Write - writing a new data set or appending data to an existing data set
+
+* Read - preparing data to be read and/or actually reading the data.
+* Write - writing a new data set or appending data to an existing data set.
 * Copy - for performing an implicit data copy on behalf of the application.  The decision to do an implicit copy is made by the control plane, typically for performance or governance reasons.
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/modules.md:44]: added - 

Delete - for deleting objects or data sets.

File: site/docs/concepts/config-policies.md [hunk 13]
```diff
@@ -20,9 +20,7 @@ Available properties:
 
 Policies are written in rego files. Each file declares a package `adminconfig`.
 
-Rules are written in the following syntax: `config[{"capability": capability, "decision": decision}]` where
-
-`capability` represents a required module capability, such as "read", "write", "transform" and "copy".
+Rules are written in the following syntax: `config[{"capability": capability, "decision": decision}]` where `capability` represents a required module capability, such as "read", "write", "transform" and "copy".
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/config-policies.md]: you are right. Initially it didn't seem right to just inline in the middle of a sentence. restored the original and added a "-" like this:

Rules are written in the following syntax: `config[{"capability": capability, "decision": decision}]` where - 

`capability` represents a required module capability, such as "read", "write", "transform" and "copy".

`decision` is a JSON structure that matches `Decision` defined above.

File: site/docs/concepts/architecture.md (hunk 6 above)
Comment (aradhalevy) [reply site/docs/concepts/architecture.md]: added a link to the optimization goals section of IT policies.

File: site/docs/concepts/architecture.md (hunk 1 above)
Comment (aradhalevy) [reply site/docs/concepts/architecture.md:13]: Done, it is now just before the Modules section.

File: site/docs/concepts/config-policies.md [hunk 14]
```diff
@@ -57,8 +55,9 @@ config[{"capability": "read", "decision": decision}] {
 `restrictions` provides restrictions for `modules`, `clusters` and `storageaccounts`.
 Each restriction provides a list or a range of allowed values for a property of module/cluster/storageaccount object. For example, to restrict a module type to either "service" or "plugin", we'll use "type" as a property, and [ "service","plugin ] as a list of allowed values.
 Properties of a module can be found inside [`FybrikModule`](../reference/crds.md#fybrikmodule) Spec.
```
[the remark this answers is not in the document]
Comment (aradhalevy) [reply site/docs/concepts/config-policies.md:59]: Done

File: site/docs/concepts/architecture.md (hunk 4 above)
Comment (aradhalevy) [reply site/docs/concepts/architecture.md]: Added the 4 options to the explanation.
changed the Policy manager to Data Policy in Connectors page.
changed in the picture and text Credential connector -> Credential management connector.
Added CSP to connectors page.

File: site/docs/concepts/architecture.md (hunk 2 above)
Comment (roytman) [reply site/docs/concepts/architecture.md:14]: Sorry for misleading. 
Let's try this way:

"Before the data user can perform any actions, a data operator has to [install](../get-started/quickstart.md) Fybrik and modules; and to configure Fybrik connectors."

Now you start the Modules sub section. 
and into it put 
[Modules](./modules.md) (M) describe capabilities that can be included in a data plane.

**Comment by roytman** [comment]
a small comment

File: site/docs/concepts/architecture.md (hunk 2 above)
Comment (aradhalevy) [reply site/docs/concepts/architecture.md]: Done

**Review by roytman — COMMENTED** [review]
@aradhalevy , if you have the Architecture picture, can you replace the Egeria (catalog) by OpenMetadata.
and maybe slightly increase the Manager rectangle, to leave place for the "Manager " text.

File: site/docs/concepts/architecture.md [hunk 15]
```diff
@@ -2,27 +2,54 @@
...
 
 ![Architecture](../static/architecture.png)
 
-Before the data user can perform any actions a data operator has to [install](../get-started/quickstart.md) Fybrik and modules. 
-[Modules](./modules.md) (M) describe capabilities that can be included in a data plane.  These may be existing open source or third party services, or custom ones.  The module of a service indicates the capabilities it supports, the formats and interfaces, and how to deploy the service.  Modules may describe externally deployed services, or services deployed by fybrik.  Examples of modules are those that provide read/write access or produce implicit copies that serve as lower latency caches of remote assets. Modules may also perform actions to enforce data governance policy decisions, such as masking or redaction as examples.
+Before the data user can perform any actions a data operator has to [install](../get-started/quickstart.md) Fybrik and modules; and to configure Fybrik connectors.
```
Comment (shlomitk1) [inline site/docs/concepts/architecture.md:13]: Configuring Fybrik connectors is a part of Fybrik installation.

File: site/docs/concepts/connectors.md [hunk 16]
```diff
@@ -50,16 +50,18 @@ An example for Vault Read Secret API call which uses the Vault parameters is as
 $ curl --header "X-Vault-Token: ..." -X GET https://<address>/<secretPath>
 ```
 
-More about `kubenertes` auth method and vault roles can be found in Vault [documentation](https://www.vaultproject.io/docs/auth/kubernetes).
+More about `kubenertes` auth method and vault roles can be found in the Vault [documentation](https://www.vaultproject.io/docs/auth/kubernetes).
```
Comment (shlomitk1) [inline site/docs/concepts/connectors.md:53]: Vault is the product name. 'the' is not required.

File: site/docs/concepts/connectors.md [hunk 17]
```diff
@@ -50,16 +50,18 @@ An example for Vault Read Secret API call which uses the Vault parameters is as
...
 
-### Policy manager
+### Data Policy
```
Comment (shlomitk1) [inline site/docs/concepts/connectors.md:55]: Why the change? This section describes Policy managers such as OPA.

File: site/docs/concepts/connectors.md [hunk 18]
```diff
@@ -50,16 +50,18 @@ An example for Vault Read Secret API call which uses the Vault parameters is as
...
 
 Data governance policies are defined externally in the data governance manager of choice. 
 
 Enforcing data governance policies requires a Policy Decision Point (PDP) that dictates what enforcement actions need to take place.
 Fybrik supports a wide and extendable set of enforcement actions to perform on data read, copy, (future) write or delete. These include transformation of data, verification of the data, and various restrictions on the external activity of an application that can access the data.
 
 A PDP returns a list of enforcement actions given a set of policies and specific context about the application and the data it uses. 
-Fybrik includes a PDP that is powered by [Open Policy Agent](https://www.openpolicyagent.org/) (OPA). However, the PDP can also use external policy managers via connectors, to cover some or even all policy types. 
+Fybrik includes a PDP that is powered by [Open Policy Agent](https://www.openpolicyagent.org/) (OPA). However, the PDP can also use external policy managers (Data policy) via connectors, to cover some or even all policy types.
```
Comment (shlomitk1) [inline site/docs/concepts/connectors.md:63]: No need for "(Data policy)".

File: site/docs/concepts/connectors.md [hunk 19]
```diff
@@ -50,16 +50,18 @@ An example for Vault Read Secret API call which uses the Vault parameters is as
...
 Enforcing data governance policies requires a Policy Decision Point (PDP) that dictates what enforcement actions need to take place.
 Fybrik supports a wide and extendable set of enforcement actions to perform on data read, copy, (future) write or delete. These include transformation of data, verification of the data, and various restrictions on the external activity of an application that can access the data.
 
 A PDP returns a list of enforcement actions given a set of policies and specific context about the application and the data it uses. 
-Fybrik includes a PDP that is powered by [Open Policy Agent](https://www.openpolicyagent.org/) (OPA). However, the PDP can also use external policy managers via connectors, to cover some or even all policy types. 
+Fybrik includes a PDP that is powered by [Open Policy Agent](https://www.openpolicyagent.org/) (OPA). However, the PDP can also use external policy managers (Data policy) via connectors, to cover some or even all policy types. 
 
+### CSP
 
+Making an optmiziation goal is optional, therefore Fybrik has a default CSP solver but supports options for others, as explained [here](./optimizer.md).
```
Comment (shlomitk1) [inline site/docs/concepts/connectors.md:67]: I don't understand this sentence.

File: site/docs/concepts/modules.md [hunk 20]
```diff
@@ -1,7 +1,7 @@
 # Modules
 
 The project currently has two extension mechanisms, namely connectors and modules. 
-This page describes what modules are and how they are leveraged by the control plane to build the data plane flow.  
+This page describes what modules are and how they are leveraged by the fybrik's control plane to build the data plane flow.
```
Comment (shlomitk1) [inline site/docs/concepts/modules.md:4]: fybrik's -> Fybrik

File: site/docs/concepts/taxonomy.md [hunk 21]
```diff
@@ -1,15 +1,20 @@
...
+Fybrik interacts with multiple external components, such as the data catalog, data governance policy manager, and modules.  In order for fybrik to orchestrate the data plane of a given workload it is essential that all the components involved use common terms.  
+For example, if a data governance policy refers to a particular transform it is crucial that the module implementing that transform refer to it in the same way, or for fybrik to be able to map between the disparate terms.
 
-* FybrikApplication yaml - information provided about the workload and the datasets
-* Fybrik manager (FybrikApplication controller)
-* Data catalog
-* Data Governance Policy Manager
-* Config Policy Manager
-* FybrikModules
+some components that use taxonomy:
```
Comment (shlomitk1) [inline site/docs/concepts/taxonomy.md:8]: some -> Some

File: site/docs/concepts/taxonomy.md [hunk 22]
```diff
@@ -1,15 +1,20 @@
...
+
+* [FybrikApplication](../architecture/#fybrikapplication) yaml - information provided about the workload and the datasets.
+* Fybrik manager ([FybrikApplication controller](../architecture/#fybrikapplication)) - validates that the data is used in accord with the data governance policies and the IT config policies.
+* [Data catalog](../connectors/#data-catalog) - provides metadata about the asset.
+* [Data Governance Policy Manager](../connectors/#policy-manager) - defines the governance policies to follow.
+* [Config Policy Manager](./config-policies.md) - defines the IT policies to follow.
+* [FybrikModules](./modules.md) - describes capabilities that can be included in a data plane.
```
Comment (shlomitk1) [inline site/docs/concepts/taxonomy.md:15]: describe

File: site/docs/concepts/architecture.md [hunk 23]
```diff
@@ -2,27 +2,54 @@
...
-could be used as a replacement. The `PlotterController` also collects statuses and distributes
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## [Connectors](./connectors.md)
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+The current abstraction supports 4 different [connectors](./connectors.md): one for data catalog (2), one for policy manager (3), one for credential management (6) and one for CSP (7). It is designed in an open way so that multiple different catalog, policy, credentials and CSP frameworks of all kinds of cloud and on-prem systems can be supported.
```
Comment (shlomitk1) [inline site/docs/concepts/architecture.md:25]: Please stay with 3 connectors and leave CSP out of it. 
There is no external connector for CSP.

File: site/docs/concepts/architecture.md [hunk 24]
```diff
@@ -2,27 +2,54 @@
...
+
```
Comment (shlomitk1) [inline site/docs/concepts/architecture.md:26]: The original text is more accurate. 
The abstraction supports two connectors: one for data catalog (2) and one for policy manager (3). 
(6) is mentioned in the next paragraph under the name "Vault API". 
As for (7), there is no external CSP connector. Optimizer invokes a CSP solver directly. The CSP solver is pluggable and can be replaced as long as it supports the same modelling language and the interface.

**Review by roytman — COMMENTED** [review]
@aradhalevy, please update the architecture.png file

File: site/docs/concepts/architecture.md [hunk 25]
```diff
@@ -2,27 +2,54 @@
...
+## [Connectors](./connectors.md)
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+ The current abstraction supports 2 different  [connectors](./connectors.md): one for data catalog and one for policy manager. It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported.
+
+The data steward configures policies  in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
+
+## FybrikApplication
+Once Fybrik is deployed and configured, including catalogs, modules and connectors, a developer can submit a `FybrikApplication` CRD to Kubernetes.
```
Comment (roytman) [inline site/docs/concepts/architecture.md:30]: a developer doesn't submit a FybrikApplication CRD, he creates (submits) an instance of the CRD. 
CRD is a **Definition** of Custom Resources, and here the developer creates a Custom Resource

File: site/docs/concepts/architecture.md [hunk 26]
```diff
@@ -2,27 +2,54 @@
...
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## [Connectors](./connectors.md)
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+ The current abstraction supports 2 different  [connectors](./connectors.md): one for data catalog and one for policy manager. It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported.
+
+The data steward configures policies  in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
+
+## FybrikApplication
```
Comment (roytman) [inline site/docs/concepts/architecture.md:29]: Maybe we should add links to FybrikApplication, Plotter and Blueprint references. It will be similar to Modules and Connectors.

Comment (aradhalevy) [reply site/docs/concepts/architecture.md]: Added

File: site/docs/concepts/architecture.md (hunk 25 above)
Comment (aradhalevy) [reply site/docs/concepts/architecture.md]: changed to " a developer can submit a `FybrikApplication`  Custom Resource to Kubernetes "

**Review by roytman — APPROVED** [review]
